### PR TITLE
feat: new BuildingSystem and some languages support

### PR DESCRIPTION
**Pull request by mtripg6666tdr**
Hi! Thank you for waiting!
Here are my changes which add
* brand new building system and you can now build:
  * TypeScript
  * TSX
  * CoffeeScript
  * Sass/Scss
* new language support for CoffeeScript

I also made some changes for preventing duplicated code and a little some code refactor (such as removing code which never runs).
Please feel free to request changes or something.
I hope this will work for you!

### CODE REVIEW

File: src/building/coffeescript.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { compile } from 'coffeescript';
+import { BaseBuildingSystem } from './base';
+
+export class CoffeeScriptBuildingSystem extends BaseBuildingSystem {
+  protected buildSnipet(content: string): Promise<string> {
```
Comment (eve0415) [inline src/building/coffeescript.ts:5]:
```suggestion
  protected buildSnippet(content: string): Promise<string> {
```

File: src/building/sass.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import sass from 'sass';
+import { BaseBuildingSystem } from './base';
+
+export class SassBuildingSystem extends BaseBuildingSystem {
+  protected async buildSnipet(content: string): Promise<string> {
```
Comment (eve0415) [inline src/building/sass.ts:5]:
```suggestion
  protected async buildSnippet(content: string): Promise<string> {
```

File: src/building/typescript.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import type { CompilerOptions } from 'typescript';
+import ts from 'typescript';
+import { BaseBuildingSystem } from './base';
+import tsconfig from '../../tsconfig.json';
+
+export class TypeScriptBuildingSystem extends BaseBuildingSystem {
+  protected buildSnipet(content: string): Promise<string> {
```
Comment (eve0415) [inline src/building/typescript.ts:7]:
```suggestion
  protected buildSnippet(content: string): Promise<string> {
```

File: src/evaluation/typescript.ts [hunk 4]
```diff
@@ -20,11 +20,4 @@ export class TypeScriptEvaluationSystem extends BaseEvaluationSystem {
       res(new JavaScriptEvaluationSystem().evaluate(script.outputText));
     });
   }
-
-  protected override createMessage(contents: unknown[]): BaseMessageOptions {
-    const processed = this.processContent(contents).filter(
-      c => !c.includes('Could not open history file') || !/^\.{2,}$/.test(c)
-    );
-    return super.createMessage(processed);
-  }
```
Comment (eve0415) [inline src/evaluation/typescript.ts:29]: これいる

Comment (mtripg6666tdr) [reply src/evaluation/typescript.ts:29]: 走らない

Comment (eve0415) [reply src/evaluation/typescript.ts:29]: 嘘

Comment (mtripg6666tdr) [reply src/evaluation/typescript.ts:29]: JavaScriptEvaluationSystemに処理が渡っちゃってるから、TypeScript側のこれは動かない

**Comment by eve0415** [comment]
scss とか prettier 走らせれる

**Comment by mtripg6666tdr** [comment]
> scss とか prettier 走らせれる

fixed :wink:

**Comment by eve0415** [comment]
たぶんおっけ、ダメだったらまたなんとかします

**Comment by eve0415** [comment]
これを基盤にして、もっと面白い機能つけます

**Pull request merged**